src/test/java/com/fasterxml/jackson/databind/deser/SkipNulls4441Test.java

```diff
@@ -0,0 +1,149 @@
+package com.fasterxml.jackson.databind.deser;
+
+import com.fasterxml.jackson.annotation.JsonSetter;
+import com.fasterxml.jackson.annotation.Nulls;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.json.JsonMapper;
+import org.junit.jupiter.api.Test;
+
+import java.beans.ConstructorProperties;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+import static com.fasterxml.jackson.databind.BaseTest.a2q;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+
+// [databind#4441] @JsonSetter(nulls = Nulls.SKIP) doesn't work in some situations
+public class SkipNulls4441Test {
+
+    static class Middle {
+        @JsonSetter(nulls = Nulls.SKIP)
+        private final List<Inner> listInner = new ArrayList<>();
+        private final String field1;
+
+        @ConstructorProperties({"field1"})
```

Comment:
Let's change this to use `@JsonCreator` instead to reduce use of `@ConstructorProperties` (in JPMS it's in extra package)